DQMOffline/L1Trigger/src/L1TTauOffline.cc

```diff
@@ -0,0 +1,677 @@
+	AntiEleInputTag_(consumes<reco::PFTauDiscriminator>(ps.getUntrackedParameter<edm::InputTag>("antiEleInputTag"))),
+	DecayModeFindingInputTag_(consumes<reco::PFTauDiscriminator>(ps.getUntrackedParameter<edm::InputTag>("decayModeFindingInputTag"))),
+	comb3TInputTag_(consumes<reco::PFTauDiscriminator>(ps.getUntrackedParameter<edm::InputTag>("comb3TInputTag"))),
+	MuonInputTag_(consumes<reco::MuonCollection>(ps.getUntrackedParameter<edm::InputTag>("muonInputTag"))),
+	MetInputTag_(consumes<reco::PFMETCollection>(ps.getUntrackedParameter<edm::InputTag>("metInputTag"))),
+	VtxInputTag_(consumes<reco::VertexCollection>(ps.getUntrackedParameter<edm::InputTag>("vtxInputTag"))),
+	BsInputTag_(consumes<reco::BeamSpot>(ps.getUntrackedParameter<edm::InputTag>("bsInputTag"))),
+        triggerEvent_(consumes < trigger::TriggerEvent > (ps.getUntrackedParameter < edm::InputTag > ("trigInputTag"))),
+	trigProcess_(ps.getUntrackedParameter<string>("trigProcess")),
+        triggerResults_(consumes < edm::TriggerResults > (ps.getUntrackedParameter < edm::InputTag > ("trigProcess_token"))),
+        triggerPath_(ps.getUntrackedParameter < vector<std::string> > ("triggerNames")),
+        histFolder_(ps.getParameter < std::string > ("histFolder")),
+        efficiencyFolder_(histFolder_ + "/efficiency_raw"),
+        stage2CaloLayer2TauToken_(consumes < l1t::TauBxCollection > (ps.getUntrackedParameter < edm::InputTag > ("l1tInputTag"))),
+        tauEfficiencyThresholds_(ps.getParameter < std::vector<double> > ("tauEfficiencyThresholds")),
+        tauEfficiencyBins_(ps.getParameter < std::vector<double> > ("tauEfficiencyBins"))
+{
+  edm::LogInfo("L1TTauOffline") << "Constructor " << "L1TTauOffline::L1TTauOffline " << std::endl;
+}
+
+//
+// -- Destructor
+//
+L1TTauOffline::~L1TTauOffline()
+{
+  edm::LogInfo("L1TTauOffline") << "Destructor L1TTauOffline::~L1TTauOffline " << std::endl;
+}
+
+//
+// -------------------------------------- beginRun --------------------------------------------
+//
+void L1TTauOffline::dqmBeginRun(const edm::Run& run, const edm::EventSetup& iSetup)
+// void L1TTauOffline::dqmBeginRun(edm::Run const &, edm::EventSetup const &)
+{
+  bool changed = true;
+  m_hltConfig.init(run,iSetup,trigProcess_,changed);
+
+  edm::LogInfo("L1TTauOffline") << "L1TTauOffline::beginRun" << std::endl;
+}
+
+//
+// -------------------------------------- bookHistos --------------------------------------------
+//
+void L1TTauOffline::bookHistograms(DQMStore::IBooker & ibooker, edm::Run const &, edm::EventSetup const &)
+{
+  edm::LogInfo("L1TTauOffline") << "L1TTauOffline::bookHistograms" << std::endl;
+
+  //book at beginRun
+  bookTauHistos(ibooker);
+ 
+  vector<string>::const_iterator trigNamesIt  = triggerPath_.begin();
+  vector<string>::const_iterator trigNamesEnd = triggerPath_.end();
+
+  for (; trigNamesIt!=trigNamesEnd; ++trigNamesIt) { 
```

Comment:
please rewrite as a range based loop

for ( auto const & trigNamesIt: triggerPath_)...